File: 240430/코딩톡/coding-talk.py
N, M, P = map(int, input().split())
messages = []
not_reads = []
for _ in range(M):
    c, u = input().split()
    messages.append(c)
    not_reads.append(int(u))

read_chk = [False] * 26
for i in range(M):
    if not_reads[P - 1] == 0:
        break

    if i >= P - 1:
        message = ord(messages[i]) - ord('A')
        read_chk[message] = True
    elif not_reads[i] == not_reads[P - 1]:
        message = ord(messages[i]) - ord('A')
        read_chk[message] = True


if not_reads[P - 1] == 0:
    print()
else:
    for i in range(N):
        if not read_chk[i]:
            print(chr(i + ord('A')), end = ' ')

File: 260407/코딩톡/coding-talk.cpp
#include <bits/stdc++.h>
using namespace std;

int main() 
{
    int n, m, p;
    cin >> n >> m >> p;

    vector<char>messages(m);
    vector<int> not_reads(m);

    for(int i = 0; i < m; i++)
        cin >> messages[i] >> not_reads[i];

    // Please write your code here.
    vector<bool> read_chk(26, false);

    for(int i = 0; i < m; i++)
    {
        if(not_reads[p - 1] == 0)
            break;

        if(i >= p - 1)
        {
            int message = messages[i] - 'A';
            read_chk[message] = true;
        }
        else if(not_reads[i] == not_reads[p - 1])
        {
            int message = messages[i] - 'A';
            read_chk[message] = true;
        }
    }

    if(not_reads[p - 1] == 0)
        cout << endl;
    else
    {
        for(int i = 0; i < n; i++)
        {
            if(!read_chk[i])
                cout << char(i + 'A') << ' ';
        }
    }
    return 0;
}